min pip version

lucylq opened the pull request:
see [T188128067](https://www.internalfb.com/intern/tasks/?t=188128067)

huydhn commented on .github/workflows/build-wheels-linux.yml:
```diff
@@ -42,6 +42,7 @@ jobs:
             post-script: build/packaging/post_build_script.sh
             smoke-test-script: build/packaging/smoke_test.py
             package-name: executorch
+            python-version: '3.10'
```
Pinning pip version is fine but I don't think `python-version` is a legit input parameter here.  The workflow is failing https://github.com/pytorch/executorch/actions/runs/9022911297

The list of supported inputs can be found here https://github.com/pytorch/test-infra/blob/main/.github/workflows/build_wheels_linux.yml#L6-L77

huydhn replied:
I have a task in my queue to add support for this though, so that different repos could specify which python versions they want instead of strictly following PyTorch build matrix, which covers all 3.8 to 3.12 atm

huydhn commented on build/packaging/pre_build_script.sh:
```diff
@@ -16,6 +16,7 @@ set -euxo pipefail
 readonly BUILD_DEPS=(
   # This list must match the build-system.requires list from pyproject.toml.
   "cmake"
+  "pip>=23"
```
In case you couldn't set this for older python versions, I wonder if this syntax works:

```
pip ; python_version < "3.10"
pip>=23 ; python_version >= "3.10"
```

https://github.com/pytorch/pytorch/blob/main/.ci/docker/requirements-ci.txt#L220-L221

Building ET with older python version is expected to fail atm IIRC

huydhn commented:
@pytorchbot rebase

lucylq committed:
min pip version (#3526)

Summary:
see [T188128067](https://www.internalfb.com/intern/tasks/?t=188128067)
and https://github.com/pytorch/torchchat/issues/726


Differential Revision: D57032452

Pulled By: lucylq
